docs/gui/screens.md

```diff
@@ -4,73 +4,241 @@ Screens are typically the base of all Graphical User Interfaces (GUIs) in Minecr
+
+`scissorArea` crops the area where the element can render. If `scissorArea` is `null`, then the entire element is rendered to the screen. Similarly, if the `scissorArea` rectangle does not intersect with the `bounds`, then nothing will be rendered.
 
-Any GUI rendered by Minecraft is typically done using `GuiGraphics`. `GuiGraphics` is the first parameter to almost all rendering methods; it contains basic methods to render commonly used objects. These fall into five categories: colored rectangles, strings, textures, items, and tooltips. There is also an additional method for rendering a snippet of a component (`#enableScissor` / `#disableScissor`). `GuiGraphics` also exposes the `PoseStack` which applies the transformations necessary to properly render where the component should be rendered. Additionally, colors are in the [ARGB][argb] format.
+The remaining three methods handles the actual rendering of the element. `pipeline` defines the shaders and metadata used by the element. `textureSetup` can specify either `Sampler0`, `Sampler1`, `Sampler2`, or some combination in the fragment shader. Finally, `buildVertices` passes the vertices to upload to the buffer. It takes in the `VertexConsumer` to pass the vertices to and the `Z` coordinate to use.
+
+NeoForge adds the method `GuiGraphics#submitGuiElementRenderState` to submit a custom element render state if the available methods provided by `GuiGraphics` is not enough.
+
+```java
+// For some GuiGraphics graphics
+graphics.submitGuiElementRenderState(new GuiElementRenderState() {
+
+    // Store the current pose of the stack
+    private final Matrix3x2f pose = new Matrix3x2f(graphics.pose());
+    // Store the current scissor area
+    @Nullable
+    private final ScreenRectangle scissorArea = graphics.peekScissorStack();
+
+    @Override
+    public ScreenRectangle bounds() {
+        // We will assume the bounds is 0, 0, 10, 10
+        
+        // Compute the initial rectangle
+        ScreenRectangle rectangle = new ScreenRectangle(
+            // The XY position
+            0, 0,
+            // The width and height of the element
+            10, 10
+        );
+
+        // Transform the rectangle to its appropriate location using the pose
+        rectangle = rectangle.transformMaxBounds(this.pose);
+
+        // If there is a scissor area defined, return the intersection of the two rectangles
+        // Otherwise, return the full bounds
+        return this.scissorArea != null
+            ? this.scissorArea.intersection(rectangle)
+            : rectangle;
+    }
+
+    @Override
+    @Nullable
+    public ScreenRectangle scissorArea() {
+        return this.scissorArea;
+    }
+
+    @Override
+    public RenderPipeline pipeline() {
+        return RenderPipelines.GUI;
+    }
+
+    @Override
+    public TextureSetup textureSetup() {
```

Comment:
It may be worth elaborating on TextureSetup and what the different Samplers mean?